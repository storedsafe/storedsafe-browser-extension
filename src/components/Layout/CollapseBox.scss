@import '../../variables';

.collapse-box {
  .collapse-box-title {
    position: relative;
    overflow: auto;
    border: 0;
    border-left: 4px solid $color-primary-0;
    background-color: $color-primary-1;
    cursor: pointer;
    transition: border-left-width $transition-duration ease-out;
  }

  .collapse-box-title {
    padding: $spacing;
  }

  .collapse-box-title::after {
    @extend %arrow;
    transition: transform $transition-duration ease-out;
  }

  .collapse-box-children {
    border: 0 solid $color-primary-0;
    border-width: 0 0 0 2px;
    margin-left: 2px;
  }

  .collapse-box-children.padded {
    padding: $spacing / 2;
  }
}

.collapse-box.collapsed {
  .collapse-box-title::after {
    transform: rotate(90deg);
  }

  .collapse-box-title {
    border-left-width: 3px;
  }
}
