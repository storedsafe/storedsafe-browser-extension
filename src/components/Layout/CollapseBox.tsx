import * as React from 'react';
import { Collapsible } from './';
import './CollapseBox.scss';

export interface CollapseBoxProps {
  startCollapsed?: boolean;
  title: React.ReactNode;
  children: React.ReactNode;
  padded?: boolean;
}

export const CollapseBox: React.FunctionComponent<CollapseBoxProps> = ({
  startCollapsed,
  title,
  children,
  padded,
}: CollapseBoxProps) => {
  const [collapsed, setCollapsed] = React.useState<boolean>(startCollapsed);

  return (
    <div className={`collapse-box${collapsed ? ' collapsed' : ''}`}>
      <div className="collapse-box-title" onClick={(): void => setCollapsed(!collapsed)}>
        {title}
      </div>
      <Collapsible collapsed={collapsed}>
        <div className={`collapse-box-children${padded ? ' padded' : ''}`}>
          {children}
        </div>
      </Collapsible>
    </div>
  );
}

CollapseBox.defaultProps = {
  startCollapsed: true,
  padded: true,
};
