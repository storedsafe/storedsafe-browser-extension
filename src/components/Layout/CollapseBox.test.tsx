import * as React from 'react';
import { shallow } from 'enzyme';
import { CollapseBox } from './CollapseBox';

test('<CollapseBox />', () => {
  const wrapper = shallow(<CollapseBox title="Title">Children</CollapseBox>);
  expect(wrapper).toMatchSnapshot();
});

test('<CollapseBox collapsed=true />', () => {
  const wrapper = shallow(<CollapseBox
    startCollapsed={true}
    title="Title">
    Children
  </CollapseBox>);
  expect(wrapper).toMatchSnapshot();
});

test('<CollapseBox collapsed=false />', () => {
  const wrapper = shallow(<CollapseBox
    startCollapsed={false}
    title="Title">
    Children
  </CollapseBox>);
  expect(wrapper).toMatchSnapshot();
});

test('<CollapseBox padded=false />', () => {
  const wrapper = shallow(<CollapseBox
    padded={false}
    title="Title">
    Children
  </CollapseBox>);
  expect(wrapper).toMatchSnapshot();
});

test('<CollapseBox padded=true />', () => {
  const wrapper = shallow(<CollapseBox
    padded={true}
    title="Title">
    Children
  </CollapseBox>);
  expect(wrapper).toMatchSnapshot();
});
