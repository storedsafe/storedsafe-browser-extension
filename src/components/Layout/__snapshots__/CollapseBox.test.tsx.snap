// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollapseBox /> 1`] = `ShallowWrapper {}`;

exports[`<CollapseBox collapsed=false /> 1`] = `ShallowWrapper {}`;

exports[`<CollapseBox collapsed=true /> 1`] = `ShallowWrapper {}`;

exports[`<CollapseBox padded=false /> 1`] = `ShallowWrapper {}`;

exports[`<CollapseBox padded=true /> 1`] = `ShallowWrapper {}`;
